src/test/java/ay2021s1_cs2103_w16_3/finesse/model/frequent/FrequentIncomeListTest.java

```diff
@@ -0,0 +1,88 @@
+
+    @Test
+    public void add_nullFrequentIncome_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> frequentIncomeList.add(null));
+    }
+
+    @Test
+    public void add_duplicateFrequentIncome_throwsDuplicateFrequentExpenseException() {
+        FrequentIncome frequentIncome = new FrequentTransactionBuilder(PART_TIME)
+                .withCategories(VALID_CATEGORY_WORK).buildFrequentIncome();
+
+        frequentIncomeList.add(frequentIncome);
+
+        FrequentIncome frequentIncomeCopy = new FrequentTransactionBuilder(frequentIncome).buildFrequentIncome();
+
+        assertThrows(DuplicateFrequentTransactionException.class, () -> frequentIncomeList.add(frequentIncomeCopy));
+    }
+
+    @Test
+    public void setFrequentIncome_nullList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, ()
+            -> frequentIncomeList.setFrequentIncomes((List<FrequentIncome>) null));
+    }
+
+    @Test
+    public void setFrequentIncomes_list_replacesOwnListWithProvidedList() {
+        frequentIncomeList.add(PART_TIME);
+        List<FrequentIncome> transactionList = Collections.singletonList(INVESTING);
+        this.frequentIncomeList.setFrequentIncomes(transactionList);
+        FrequentIncomeList expectedTransactionList = new FrequentIncomeList();
+        expectedTransactionList.add(INVESTING);
+        assertEquals(expectedTransactionList, this.frequentIncomeList);
+    }
+
+    @Test
+    public void asUnmodifiableObservableList_modifyList_throwsUnsupportedOperationException() {
+        assertThrows(UnsupportedOperationException.class, ()
+            -> frequentIncomeList.asUnmodifiableObservableList().remove(0));
+    }
+
+    @Test
+    public void equals_distinctFrequentIncomeListsWithSameAttributes_returnsTrue() {
+        List<FrequentIncome> frequentIncomes = getTypicalFrequentIncome();
+        FrequentIncomeList firstTransactionList = new FrequentIncomeList();
+        FrequentIncomeList secondTransactionList = new FrequentIncomeList();
+
+        firstTransactionList.setFrequentIncomes(frequentIncomes);
+        secondTransactionList.setFrequentIncomes(frequentIncomes);
+
+        assertNotSame(firstTransactionList, secondTransactionList);
+        assertEquals(firstTransactionList, secondTransactionList);
+    }
+
+    @Test
+    public void hashCode_distinctFrequentIncomeListsWithSameAttributes_returnsTrue() {
+        List<FrequentIncome> frequentIncomes = getTypicalFrequentIncome();
+        FrequentIncomeList firstTransactionList = new FrequentIncomeList();
+        FrequentIncomeList secondTransactionList = new FrequentIncomeList();
+
+        firstTransactionList.setFrequentIncomes(frequentIncomes);
+        secondTransactionList.setFrequentIncomes(frequentIncomes);
+
+        assertNotSame(firstTransactionList, secondTransactionList);
+        assertEquals(firstTransactionList.hashCode(), secondTransactionList.hashCode());
+    }
+
+}
```

Comment:
Shouldn't this test mirror `FrequentExpenseListTest`? There are less tests here than in `FrequentExpenseListTest`.

Reply:
The reason for this is because I have only added the `AddFrequentIncomeCommand` hence I did not write the other methods within `FrequentIncomeList` yet. Therefore I did not write the tests for the other methods.